DQM/SiPixelPhase1Track/plugins/SiPixelPhase1TrackResiduals.cc

```diff
@@ -81,6 +92,13 @@ namespace {
         histo[RESIDUAL_X].fill(it.resXprime, id, &iEvent);
         histo[RESIDUAL_Y].fill(it.resYprime, id, &iEvent);
 
+        if (it.resXprimeErr != 0 && it.resXprimeErr != 0) {
+          histo[NormRes_X].fill(it.resXprime / it.resXprimeErr, id, &iEvent);
+          histo[NormRes_Y].fill(it.resYprime / it.resYprimeErr, id, &iEvent);
+          histo[DRNR_X].fill(it.resXprime / it.resXprimeErr, id, &iEvent);
```

Comment:
how is `DRNR_X` different from `NormRes_X` ?

Reply:
No difference at this level, the differences are introduced in the harvesting step. I separate the plots here in order to have a proper histogram name on the DQMIO root file for DRnR. Summarizing: here DRNR_X is used to fill a 2D Profile which will show NormRes mean value for each bin, in the harvesting the content of this 2D profile is changed replacing the Mean value with RMS for each bin(module).

Reply:
Can't the histogram be booked directly in the harvesting step?

Reply:
In principle we can have the MEs booked and filled in the step3 but to do that we need to implement a method top evaluate RMS of single module in a running way and be sure to be thread and Concurrent LS safe. In the way things are implemented the 2DProfile ME take care of everything and the "cost" is to have an apparently duplicated entry on step3.